src/dsp_tools/xmllib/models/resource.py

```diff
@@ -311,21 +313,39 @@ def add_simpletext_optional(
     #######################
 
     def add_richtext(
-        self, value: str, prop_name: str, permissions: str | None = None, comment: str | None = None
+        self,
+        value: str,
+        prop_name: str,
+        permissions: str | None = None,
+        comment: str | None = None,
+        newline_replacement: NewlineReplacementTags = NewlineReplacementTags.newline,
     ) -> Resource:
+        value = replace_newlines_with_tags(str(value), newline_replacement)
         self.values.append(Richtext(value, prop_name, permissions, comment, self.res_id))
         return self
 
     def add_richtexts(
-        self, values: list[str], prop_name: str, permissions: str | None = None, comment: str | None = None
+        self,
+        values: list[str],
+        prop_name: str,
+        permissions: str | None = None,
+        comment: str | None = None,
+        newline_replacement: NewlineReplacementTags = NewlineReplacementTags.newline,
     ) -> Resource:
+        values = [replace_newlines_with_tags(str(x), newline_replacement) for x in values]
```

Comment:
stylistic nitpick: I don't really like the use of variable names like `x` because they deliberately do not say what the variable means. In cases like this, I would prefer `[... for value in values]` or for brevity's sake even better `[... for v in values]`.